internal/session/handle_capability.go

```diff
@@ -3,15 +3,31 @@ package session
 import (
 	"context"
 
+	"github.com/ProtonMail/gluon/imap"
 	"github.com/ProtonMail/gluon/internal/parser/proto"
 	"github.com/ProtonMail/gluon/internal/response"
 )
 
+func (s *Session) getCaps() []imap.Capability {
+	if s.state != nil {
```

Comment:
normally we protect the `s.state` with `s.userLock` (which should probably be called stateLock now). Can you try to lock that here too? That might conflict with existing locks though, you might have to be a bit more clever.

Reply:
The lock is no longer necessary at the session level. Everything that happens at the session is single threaded now (barring any unexpected go-routines)

Reply:
Great, then perhaps we should remove it. But we'll have to be mindful once we allow multiple in-flight commands per session to put some kind of state lock back here.